Appstore screenshot: Update copies and screenshots

itsmeichigo opened the pull request:
### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
As per request from the marketing team (p91TBi-9eM-p2), this PR updates the automation scripts to take requested screenshots:
- Capture the monthly revenue: switch to capture the This Month tab of the My Store screen with mock data for the monthly stats.
- Update script to capture order creation form and product creation bottom sheet.
- Also update copies for app description

Updates for other screenshots will be handled in subsequent PRs.

### Testing instructions
<!-- Step by step testing instructions. When necessary break out individual scenarios that need testing, consider including a checklist for the reviewer to go through. -->
- Turn off the feature flag `.splitViewInOrdersTab` in `DefaultFeatureFlagService`.
- Select target WooCommerceScreenshots. Run the `testScreenshots()` UI test.
- Confirm that the test succeeds.

For copy updates, these can be checked by comparing with the Google doc in p91TBi-9eM-p2.

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

itsmeichigo committed:
Add new functions for navigating to add product

itsmeichigo committed:
new screenshots for order creation and adding products

itsmeichigo committed:
Update screenshot contents

itsmeichigo committed:
Dismiss order creation with fallback option top pop the navigation stack

itsmeichigo committed:
Switch to capturing monthly revenue

itsmeichigo committed:
Silence hound

itsmeichigo committed:
Fix issue mocking monthly stop earner stats

itsmeichigo committed:
Update mock order total

itsmeichigo committed:
Update app store description

itsmeichigo committed:
Remove tapOutside method

jaclync commented on Yosemite/Yosemite/Model/Mocks/Graphs/ScreenshotsObjectGraph.swift:
```diff
@@ -183,6 +183,26 @@ struct ScreenshotObjectGraph: MockObjectGraph {
+        createTopEarningItem(product: Products.coloradoShades, quantity: 5),
+    ])
+
     var thisYearVisitStats: SiteVisitStats {
```
nit: do we want to also delete the `thisYear*` code since they are not used anymore?

itsmeichigo replied:
I wasn't sure about this, but now that you mentioned it, I think it makes sense to clear the unused code. Removed in e06d5a7e9d87b71383834e606788a28eaa21c7c4.

jaclync commented on WooCommerce/UITestsFoundation/Screens/MyStore/PeriodStatsTable.swift:
```diff
@@ -11,12 +11,17 @@ public final class PeriodStatsTable: ScreenObject {
     private var daysTab: XCUIElement { daysTabGetter(app) }
     private var weeksTab: XCUIElement { weeksTabGetter(app) }
+    private var monthsTab: XCUIElement { monthsTabGetter(app) }
     private var yearsTab: XCUIElement { yearsTabGetter(app) }
```
nit: these variables sound a bit misleading, since both "This Month" and "This Week" tabs show stats for days. It's fine to leave them though if the changes are too big 😓 
```suggestion
    private var todayTab: XCUIElement { daysTabGetter(app) }
    private var thisWeekTab: XCUIElement { weeksTabGetter(app) }
    private var thisMonthTab: XCUIElement { monthsTabGetter(app) }
    private var thisYearTab: XCUIElement { yearsTabGetter(app) }
```

itsmeichigo replied:
Agreed! Updated in f6258115b180c19216a2f834a977d3d6b0bb7967.

jaclync commented on Yosemite/Yosemite/Model/Mocks/Graphs/ScreenshotsObjectGraph.swift:
```diff
@@ -191,7 +211,7 @@ struct ScreenshotObjectGraph: MockObjectGraph {
                     granularity: .month,
                     periodDate: date.yearStart.addingMonths(monthIndex),
-                    visitors: Int.random(in: 100 ... 1000)
+                    visitors: Int.random(in: 100 ... 500)
```
nit: do we still need the "this year" stats?

itsmeichigo replied:
Also removed in e06d5a7e9d87b71383834e606788a28eaa21c7c4.

jaclync commented on fastlane/appstoreres/metadata/source/description.txt:
```diff
@@ -1,22 +1,22 @@
 WooCommerce is the world’s most customizable open-source eCommerce platform. Whether you’re launching a business, taking brick-and-mortar retail online, or developing sites for clients, use WooCommerce for a store that powerfully blends content and commerce.
 
-Requirements: WooCommerce v3.5+, the Jetpack plugin.
+Requirements: WooCommerce v3.5+. Jetpack is also required to receive notifications.
```
amateur note: `also` sounds like Jetpack is also a requirement, I wonder if removing the `also` and mentioning the plugins with JCP might help.

itsmeichigo replied:
Thank you! I forwarded the suggestion to marketing in p91TBi-9eM-p2#comment-10824, I'll wait for the final copy to update this if needed.

jaclync reviewed (approved):
Works great, thanks for taking on the screenshot updates! 🚀 

Do you know how to find the screenshots taken in the test from Xcode @itsmeichigo? I tried following the guide in p91TBi-8mI-p2, but it's taking a very long time to generate.

itsmeichigo committed:
Remove mock data for This Year tab

itsmeichigo committed:
Rename tab-related properties

itsmeichigo commented:
Thank you @jaclync for the review! I've addressed the issues you found, and forwarded the copy suggestion to marketing. 

> Do you know how to find the screenshots taken in the test from Xcode

Sorry for not including this in the PR description 😅. You can find the screenshot in path `~/Library/Caches/tools.fastlane/screenshots` - it's likely that the `screenshots` folder isn't there yet so you may need to create one manually.

jaclync reviewed (approved):
Yay I see the screenshots from the test in `~/Library/Caches/tools.fastlane/screenshots` now! nice work :shipit:

itsmeichigo committed:
Remove Jetpack requirement from app store description

itsmeichigo commented:
FYI I removed the Jetpack requirement from the description following marketing's confirmation in ba9cbd74384e68352ca0106a6ac5d091489cb4fd. Proceeding to merge now.
